### PR TITLE
Move some namespaces on repository

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -69,9 +69,7 @@ Then you just have to implement `SyliusRepositoryInterface` and use `ResourceRep
 ```php
 namespace App\Repository;
 
-use App\Entity\Book;
-use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
+use App\Entity\Book;use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;
```
Comment (diimpp) [inline docs/configuration.md]:
```suggestion
use App\Entity\Book;
use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;
```

File: docs/configure_your_operations.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ There are some basic CRUD operations and more.
 ```php
 namespace App\Entity;
 
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Resource\Model\ResourceInterface;
 use Sylius\Resource\Metadata\AsResource;
 use Sylius\Resource\Metadata\Index;
```
Comment (diimpp) [inline docs/configure_your_operations.md]:
```suggestion

use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Metadata\Index;
use Sylius\Resource\Model\ResourceInterface;
```

Should be sorted alphabetically for this entry and others in the file

File: docs/configure_your_resource.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ It will configure your entity as a Sylius resource.
 ```php
 namespace App\Entity;
 
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Resource\Model\ResourceInterface;
 use Sylius\Resource\Metadata\AsResource;
```
Comment (diimpp) [inline docs/configure_your_resource.md]:
```suggestion
use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Model\ResourceInterface;
```

For this and other entries in this file

File: docs/create_new_resource.md [hunk 4]
```diff
@@ -136,11 +136,7 @@ Your repository should look like this:
 
 namespace App\Repository;
 
-use App\Entity\Book;
-use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
-use Doctrine\Persistence\ManagerRegistry;
-use Sylius\Bundle\ResourceBundle\Doctrine\ORM\ResourceRepositoryTrait;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
+use App\Entity\Book;use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;use Doctrine\Persistence\ManagerRegistry;use Sylius\Bundle\ResourceBundle\Doctrine\ORM\ResourceRepositoryTrait;use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;
```
Comment (diimpp) [inline docs/create_new_resource.md]:
```suggestion
use App\Entity\Book;
use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
use Doctrine\Persistence\ManagerRegistry;
use Sylius\Bundle\ResourceBundle\Doctrine\ORM\ResourceRepositoryTrait;
use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;
```

File: docs/redirect.md [hunk 5]
```diff
@@ -30,7 +30,7 @@ declare(strict_types=1);
 
 namespace App\Entity;
 
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Resource\Model\ResourceInterface;
 use Sylius\Resource\Metadata\AsResource;
 use Sylius\Resource\Metadata\Create;
 use Sylius\Resource\Metadata\Update;
```
Comment (diimpp) [inline docs/redirect.md]:
```suggestion
use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Metadata\Create;
use Sylius\Resource\Metadata\Update;
use Sylius\Resource\Model\ResourceInterface;
```
For this and other entries in this file

File: docs/resource_factories.md [hunk 6]
```diff
@@ -95,7 +95,7 @@ declare(strict_types=1);
 
 namespace App\Entity\Book;
 
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Resource\Model\ResourceInterface;
 use Sylius\Resource\Metadata\AsResource;
 use Sylius\Resource\Metadata\Create;
```
Comment (diimpp) [inline docs/resource_factories.md]:
```suggestion
use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Metadata\Create;
use Sylius\Resource\Model\ResourceInterface;
```

For this and other entries in this file

File: docs/resource_factories.md [hunk 7]
```diff
@@ -128,9 +128,7 @@ declare(strict_types=1);
 
 namespace App\Factory;
 
-use App\Entity\Book;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
-use Sylius\Resource\Factory\FactoryInterface;
+use App\Entity\Book;use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;use Sylius\Resource\Factory\FactoryInterface;
```
Comment (diimpp) [inline docs/resource_factories.md]:
```suggestion
use App\Entity\Book;
use Sylius\Resource\Doctrine\Persistence\RepositoryInterface;
use Sylius\Resource\Factory\FactoryInterface;
```

File: docs/resource_factories.md [hunk 8]
```diff
@@ -245,7 +243,7 @@ use Sylius\Resource\Metadata\Create;
 #[Create(
     factory: [BookFactory::class, 'create'], 
 )]
-class Book implements \Sylius\Component\Resource\Model\ResourceInterface
+class Book implements \Sylius\Resource\Model\ResourceInterface
```
Comment (diimpp) [inline docs/resource_factories.md]:
```suggestion
use App\Factory\BookFactory;
use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Metadata\Create;
use Sylius\Resource\Model\ResourceInterface

#[AsResource]
#[Create(
    factory: [BookFactory::class, 'create'], 
)]
class Book implements ResourceInterface
```

File: docs/responders.md [hunk 9]
```diff
@@ -66,7 +66,7 @@ Use it on your operation.
 
 namespace App\Entity;
 
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Resource\Model\ResourceInterface;
 use Sylius\Resource\Metadata\AsResource;
 use Sylius\Resource\Metadata\Show;
```
Comment (diimpp) [inline docs/responders.md]:
```suggestion
use Sylius\Resource\Metadata\AsResource;
use Sylius\Resource\Metadata\Show;
use Sylius\Resource\Model\ResourceInterface;
```

File: psalm.xml [hunk 10]
```diff
@@ -192,7 +196,9 @@
                 <directory name="src/Component/Factory" />
                 <directory name="src/Component/Metadata" />
                 <directory name="src/Component/Generator" />
+                <directory name="src/Component/Model" />
```
Comment (diimpp) [inline psalm.xml]:
```suggestion
                <directory name="src/Component/Generator" />
                <directory name="src/Component/Metadata" />
                <directory name="src/Component/Model" />
```

File: psalm.xml [hunk 11]
```diff
@@ -220,6 +226,8 @@
...
```
Comment (diimpp) [inline psalm.xml]:
```suggestion
                <directory name="src/Component/Generator" />
                <directory name="src/Component/Metadata" />
                <directory name="src/Component/Model" />
```

File: src/Component/src/Doctrine/Persistence/Exception/ExistingResourceException.php [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Doctrine\Persistence\Exception;
+
+class ExistingResourceException extends \Exception
```
Comment (diimpp) [inline src/Component/src/Doctrine/Persistence/Exception/ExistingResourceException.php]: Is it good time to rename\adjust exception?

```suggestion
class ResourceExistsException extends \RuntimeException implements ExceptionInterface
```

```php
namespace Sylius\Resource\Doctrine\Persistence\Exception;

interface ExceptionInterface extends \Sylius\Resource\Exception\ExceptionInterface extends \Throwable
``` 

https://learn.microsoft.com/en-us/java/api/com.azure.core.exception.resourceexistsexception?view=azure-java-stable

**Review by diimpp — COMMENTED** [review]
I've checked around half, will re-read after rebase.

File: src/Component/src/Doctrine/Persistence/Exception/ExistingResourceException.php (hunk 12 above)
Comment (loic425) [reply src/Component/src/Doctrine/Persistence/Exception/ExistingResourceException.php]: done, thanks a lot.

File: docs/configure_your_operations.md [hunk 13]
```diff
@@ -35,9 +35,10 @@ There are some basic CRUD operations and more.
 ```php
 namespace App\Entity;
 
-use Sylius\Resource\Model\ResourceInterface;
+
```
Comment (diimpp) [inline docs/configure_your_operations.md]:
```suggestion
```

File: docs/configure_your_resource.md
Comment (diimpp) [inline docs/configure_your_resource.md:1]: Line 127
```patch
--- | library | APagerfanta\Pagerfanta
+++ | library | Pagerfanta\Pagerfanta
```

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+use Sylius\Resource\Model\ResourceInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+use Webmozart\Assert\Assert;
+
+class InMemoryRepository implements RepositoryInterface
+{
+    protected PropertyAccessor $accessor;
+
+    protected \ArrayObject $arrayObject;
```
Comment (diimpp) [inline src/Component/src/Doctrine/Persistence/InMemoryRepository.php:30]: ArrayObject is imported at `use` section, but here specified globally

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Doctrine\Persistence;
+
+use ArrayObject;
```
Comment (diimpp) [inline src/Component/src/Doctrine/Persistence/InMemoryRepository.php]: According to symfony coding style, global classes\exceptions shouldn't be imported and used directly.

```suggestion
```

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+        /** @var array $interfaceInterfaces */
+        $interfaceInterfaces = class_implements($interface);
+
+        if (!in_array(ResourceInterface::class, $interfaceInterfaces, true)) {
+            throw new UnexpectedTypeException($interface, ResourceInterface::class);
+        }
+
+        $this->interface = $interface;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+        $this->arrayObject = new ArrayObject();
```
Comment (diimpp) [inline src/Component/src/Doctrine/Persistence/InMemoryRepository.php]:
```suggestion
        $this->arrayObject = new \ArrayObject();
```

File: src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace spec\Sylius\Resource\Doctrine\Persistence\Exception;
+
+use PhpSpec\ObjectBehavior;
+
+final class ResourceExistsExceptionSpec extends ObjectBehavior
+{
+    function it_extends_exception(): void
+    {
+        $this->shouldHaveType(\Exception::class);
```
Comment (diimpp) [inline src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php:23]: Should it be more specific?
```suggestion
        $this->shouldHaveType(\RuntimeException::class);
```

File: src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
```
Comment (diimpp) [inline src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php:25]: Wdyt adding `it_implements_exception_interface` for `src/Component/src/Doctrine/Persistence/Exception/ExceptionInterface.php` as it's important part.

Comment (loic425) [reply src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php:25]: yep, done.

File: src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php (hunk 17 above)
Comment (loic425) [reply src/Component/tests/spec/Doctrine/Persistence/Exception/ResourceExistsExceptionSpec.php:23]: I added the two assertions, cause the moved one was extending exception one.

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php (hunk 16 above)
Comment (loic425) [reply src/Component/src/Doctrine/Persistence/InMemoryRepository.php]: done

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php (hunk 14 above)
Comment (loic425) [reply src/Component/src/Doctrine/Persistence/InMemoryRepository.php:30]: yep, I've removed the "use" section.

File: src/Component/src/Doctrine/Persistence/InMemoryRepository.php (hunk 15 above)
Comment (loic425) [reply src/Component/src/Doctrine/Persistence/InMemoryRepository.php]: done

**Review by diimpp — APPROVED** [review]

File: docs/resource_factories.md [hunk 19]
```diff
@@ -245,7 +245,16 @@ use Sylius\Resource\Metadata\Create;
 #[Create(
     factory: [BookFactory::class, 'create'], 
 )]
-class Book implements \Sylius\Resource\Model\ResourceInterface
+use App\Factory\BookFactory;
```
Comment (diimpp) [inline docs/resource_factories.md]: Double `use` from lines 240-247 to this snippet.

Comment (loic425) [reply docs/resource_factories.md]: done

**Review by diimpp — APPROVED** [review]

**Review by oallain — APPROVED** [review]
Not easy to read but it's seen to be good 🤩

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Loïc! :1st_place_medal:

**Pull request merged**